src/ripple/conditions/impl/utils.h

```diff
@@ -31,338 +33,195 @@
-        *out++ = static_cast<std::uint8_t>(len & 0xFF);
-        return out;
-    }
+        if (s.empty())
+        {
+            ec = error::generic;
+            return p;
+        }
 
-    if (len <= 0xFFFFFF)
-    {
-        *out++ = 0x83;
-        *out++ = static_cast<std::uint8_t>((len >> 16) & 0xFF);
-        *out++ = static_cast<std::uint8_t>((len >> 8) & 0xFF);
-        *out++ = static_cast<std::uint8_t>(len & 0xFF);
-        return out;
+        p.tag = p.tag * 128 + s[0];
+        s += 1;
     }
 
-    if (len <= 0xFFFFFFFF)
+    if (s.empty())
     {
-        *out++ = 0x84;
-        *out++ = static_cast<std::uint8_t>((len >> 24) & 0xFF);
-        *out++ = static_cast<std::uint8_t>((len >> 16) & 0xFF);
-        *out++ = static_cast<std::uint8_t>((len >> 8) & 0xFF);
-        *out++ = static_cast<std::uint8_t>(len & 0xFF);
-        return out;
+        ec = error::generic;
+        return p;
     }
 
-    // Note: OER can represent lengths up to (2^1016) - 1,
-    // which is, truly, enough for everyone. We never
-    // exceed 2^32.
-    Throw<std::length_error>("overlong encoding length: " + std::to_string(len));
-}
+    p.length = s[0];
+    s += 1;
 
-// A "streambuf" would serve us better here - instead of the
-// crazy paired return, we consume data from it and things
-// just magically work.
-template <class InputIt>
-std::pair<InputIt, std::size_t>
-decode_length (InputIt begin, InputIt end)
-{
-    if (begin == end)
-        Throw<std::length_error>("empty buffer");
-
-    std::size_t bytes = *begin++;
+    if (p.length & 0x80)
+    { // Long form length:
+        std::size_t const cnt = p.length & 0x7F;
 
-    if (bytes < 128)
-        return { begin, bytes };
+        if (cnt == 0)
+        {
+            ec = error::generic;
+            return p;
+        }
 
-    bytes &= 0x7F;
+        if (cnt > sizeof(std::size_t))
+        {
+            ec = error::generic;
+            return p;
+        }
 
-    if (bytes > 4)
-        Throw<std::length_error>("overlong encoded length: " + std::to_string(bytes));
+        // Account for the first two bytes in the preamble:
+        if (cnt >= s.size())
+        {
+            ec = error::generic;
+            return p;
+        }
 
-    std::size_t len = 0;
+        p.length = 0;
 
-    if (std::distance (begin, end) < bytes)
-        Throw<std::length_error>("short encoded length: " + std::to_string(bytes));
+        for (std::size_t i = 0; i != cnt; ++i)
+            p.length = (p.length << 8) + s[i];
 
-    while (bytes--)
-        len = (len << 8) | *begin++;
+        s += cnt;
 
-    return { begin, len };
-}
-
-/** Encode a fixed-size octet string: OER 2.6 (2) */
-template <class InputIt, class OutputIt>
-OutputIt
-encode_octetstring(InputIt begin, InputIt end, OutputIt out)
-{
-    while (begin != end)
-        *out++ = *begin++;
+        if (p.length == 0)
+        {
+            ec = error::generic;
+            return p;
+        }
+    }
 
-    return out;
+    ec = {};
```

Comment:
I _think_ when there's no error it's usual to just leave the `error_code` alone.